timemachine/parser.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional, Dict, Any, List
+from dataclasses import dataclass, fields
+from pathlib import Path
+
+from yaml import safe_load
+
+ROOT_DIR = Path(__file__).absolute().parent
+
+MAP_GENERATION = "map_generation"
+
+
+@dataclass
+class MapGenerationConfig:
+
+    protein: str
+    ligands: List[str]
+    transformation_threshold: int = 3
+    atom_mapping_strategy: str = "geometry"
+    forcefield: str = str(ROOT_DIR.joinpath("ff/params/smirnoff_1_1_0_ccc.py"))
+    output: Optional[str] = None
+    cores: Optional[List[Dict]] = None
+    networks: Optional[List[Dict]] = None
```

Comment:
This looks good to me, although I think we might adjust a bit later if these settings should be nested

I think the top-level things that we want to specify here are:
* `protein` target
* small molecule `forcefield`
* `ligands` list
* `output` path
* `atom_mapping_strategy` -- all information required to specify a function `generate_atom_map(mol_a, mol_b) -> atom_map`. (In this case, something like "geometry" with keyword argument "threshold = 0.5".)
* `edge_selection_strategy` -- all information required to specify a function `generate_edges(ligands) -> list_of_pairs_of_ligands`.
   * Basic case is "generate a star, including (reference_ligand, other_ligand) pairs, where reference_ligand is selected from `ligands` either manually or using a named strategy."
   * We may want to apply "pre-filters" to the set of ligands, and  "post-filters" to the set of candidate edges.
      * A pre-filter applied in this case is something like "ligands must match user-specified core smarts".
      * Post-filters might include "an edge may add or remove no more than `transformation_size_threshold` atoms", and "an edge may not change ring sizes".
    * Edge selection strategies may be combined (e.g. if you have a complete strategy tailored to one core, and a complete strategy tailored to another core, we may want to apply them both to a ligand set and take the union the edges they select).